src/Utilities/Compiler/Extensions/ISymbolExtensions.cs

```diff
@@ -747,18 +747,13 @@ public static bool IsConst([NotNullWhen(returnValue: true)] this ISymbol? symbol
         }
 
         public static bool IsReadOnly([NotNullWhen(returnValue: true)] this ISymbol? symbol)
-        {
-            return symbol switch
+            => symbol switch
             {
                 IFieldSymbol field => field.IsReadOnly,
-
                 IPropertySymbol property => property.IsReadOnly,
-
-                // TODO: IMethodSymbol and ITypeSymbol also have IsReadOnly in Microsoft.CodeAnalysis 3.x
-                //       Add these cases once we move to the required Microsoft.CodeAnalysis.nupkg.
-
+                IMethodSymbol method => method.IsReadOnly,
```

Comment:
You may need to wrap this around preprocessor symbol checks for code analysis V3 or later

Reply:
Oh didn't see this was failing. That makes sense, some analyzers are still based on old version of roslyn.